airbyte-workers/src/main/java/io/airbyte/workers/DefaultReplicationWorker.java

```diff
@@ -68,20 +68,32 @@ public class DefaultReplicationWorker implements ReplicationWorker {
   private final ExecutorService executors;
   private final AtomicBoolean cancelled;
   private final AtomicBoolean hasFailed;
+  private final RecordSchemaValidator recordSchemaValidator;
 
   public DefaultReplicationWorker(final String jobId,
                                   final int attempt,
                                   final AirbyteSource source,
                                   final AirbyteMapper mapper,
                                   final AirbyteDestination destination,
                                   final MessageTracker messageTracker) {
+    this(jobId, attempt, source, mapper, destination, messageTracker, new RecordSchemaValidator());
+  }
+
+  public DefaultReplicationWorker(final String jobId,
```

Comment:
We should add a `@VisibleForTesting` annotation to this constructor and remove the `public` keyword to make this [package-private](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html), since we only expect this one to be used for testing